cmake/loki_transform.cmake

```diff
@@ -198,320 +42,59 @@ endmacro()
-        DEPENDS ${_PAR_SOURCE} ${_PAR_DEPENDS}
-        COMMENT "[clawfc] Pre-processing: target=${_PAR_TARGET} directive=${_PAR_DIRECTIVE}"
-    )
-
-endfunction()
-
-
-##############################################################################
-# .rst:
-#
-# generate_xmod
-# =============
-#
-# Call OMNI's F_Front on a file to generate its xml-parse tree and, as a
-# side effect, xmod-file.::
-#
-#   generate_xmod(
-#       OUTPUT <xml-file>
-#       SOURCE <source>
-#       [XMOD <xmod-dir1> [<xmod-dir2> ...]]
-#       [DEPENDS <dependency1> [<dependency2> ...]]
-#   )
-#
-# Note that the xmod-file will be located in the first path given to ``XMOD``.
-#
-##############################################################################
-function( generate_xmod )
-
-    set( options )
-    set( oneValueArgs SOURCE OUTPUT )
-    set( multiValueArgs XMOD DEPENDS )
-
-    cmake_parse_arguments( _PAR "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
+    set( _ARGS ${_PAR_COMMAND} )
 
     if( NOT _PAR_OUTPUT )
-        ecbuild_critical( "No OUTPUT given for generate_xmod()" )
+        ecbuild_critical( "No OUTPUT specified for loki_transform()" )
     endif()
 
-    if( NOT _PAR_SOURCE )
-        ecbuild_critical( "No SOURCE given for generate_xmod()" )
+    if( NOT _PAR_DEPENDS )
+        ecbuild_critical( "No DEPENDS specified for loki_transform()" )
     endif()
 
-    set( _ARGS )
-    list( APPEND _ARGS -fleave-comment )
+    # Translate function args to arguments for loki-transform.py
+    _loki_transform_parse_args()
 
-    if( _PAR_XMOD )
-        foreach( XMOD ${_PAR_XMOD} )
-            list( APPEND _ARGS -M ${XMOD} )
-        endforeach()
-    endif()
+    # Translate function options to arguments for loki-transform.py
+    _loki_transform_parse_options()
 
-    if( TARGET clawfc )
-        get_target_property( _CLAWFC_EXECUTABLE clawfc IMPORTED_LOCATION )
-        get_filename_component( _CLAWFC_LOCATION ${_CLAWFC_EXECUTABLE} DIRECTORY )
-        set( _F_FRONT_EXECUTABLE ${_CLAWFC_LOCATION}/F_Front )
-        list( APPEND _PAR_DEPENDS clawfc )
-    else()
-        set( _F_FRONT_EXECUTABLE F_Front )
-    endif()
+    # Ensure transformation script and environment is available
+    _loki_transform_env_setup()
+
+    ecbuild_debug( "COMMAND ${_LOKI_TRANSFORM} ${_ARGS}" )
 
     add_custom_command(
         OUTPUT ${_PAR_OUTPUT}
-        COMMAND ${_F_FRONT_EXECUTABLE} ${_ARGS} -o ${_PAR_OUTPUT} ${_PAR_SOURCE}
-        DEPENDS ${_PAR_SOURCE} ${_PAR_DEPENDS}
-        COMMENT "[OMNI] Pre-processing: ${_PAR_SOURCE}"
+        COMMAND ${_LOKI_TRANSFORM} ${_ARGS}
+        DEPENDS ${_PAR_DEPENDS} ${_LOKI_TRANSFORM_DEPENDENCY}
+        COMMENT "[Loki] Pre-processing: command=${_PAR_COMMAND} mode=${_PAR_MODE} directive=${_PAR_DIRECTIVE} frontend=${_PAR_FRONTEND}"
     )
 
 endfunction()
 
-##############################################################################
-
-macro( _loki_transform_parse_target_args _func_name )
```

Comment:
EC-phys regression tells me that there's still a dangling call to this, but it seems to be removed entirely. Was this intended to be moved to helpers, or is it entirely redundant now? In the latter case `loki_transform_target` needs a small update.

Reply:
Thanks for testing and flagging! Yes, we need regression testing of this entry point asap. Please have another go, I have removed the dangling call which shouldn't have provided more than argument checking in this instance.